skynet/agent.py

```diff
@@ -1,15 +1,23 @@
 import asyncio
 import math
 
+from fastapi import FastAPI, Request
+from pydantic import BaseModel
+
 from skynet.env import whisper_max_connections
 
 from skynet.logs import get_logger
-from skynet.modules.monitoring import CONNECTIONS_METRIC
+from skynet.modules.monitoring import CONNECTIONS_METRIC, TRANSCRIBE_GRACEFUL_SHUTDOWN, TRANSCRIBE_STRESS_LEVEL_METRIC
 
 log = get_logger(__name__)
+app = FastAPI()
```

Comment:
Since we are mounting a new app, I reckon we'd want to contain it in its own file perhaps, rather than together with this agent code?

Reply:
They need to communicate with one another (that was initially the reason for the temp file as they were in separate modules). I dropped the temp file as they can share a global. If the autoscaler sets the app's state to **drain**, that state needs to be reflected in the HAProxy agent response. So the LB can stop sending new connections to this machine. No need to stop accepting connections from Skynet itself as HAProxy will handle it.